File: api/schemas/chat/chat_stream_token_event.py
from typing import Literal
from uuid import UUID

from pydantic import BaseModel

from api import enums


class ChatStreamTokenEvent(BaseModel):
    content: str
    session_id: UUID
    type: Literal[enums.chat.ChatStreamEventTypeEnum.TOKEN] = enums.chat.ChatStreamEventTypeEnum.TOKEN

File: web/types/chat/ChatStreamTokenEvent.ts
// enums
import { ChatStreamEventTypeEnum } from '@/enums/chat';

// types
import type BaseChatStreamEvent from './BaseChatStreamEvent';

interface ChatStreamTokenEvent extends BaseChatStreamEvent {
  content: string;
  type: ChatStreamEventTypeEnum.Token;
}

export default ChatStreamTokenEvent;
